oauthproxy.go

```diff
@@ -490,6 +493,12 @@ func (p *OAuthProxy) GetRedirect(req *http.Request) (redirect string, err error)
 	}
 
 	redirect = req.Header.Get("X-Auth-Request-Redirect")
+	logger.Printf("X-Auth-Request-Redirect: %s", redirect)
+	// traefik does not send X-Auth-Request-Redirect, so give the traefik headers a try in case X-Auth-Request-Redirect is not set
+	if !p.IsValidRedirect(redirect) {
```

Comment:
This should be after the `rd` querystring parameter codepath since that is the more standard and currently adopted path.  This ordering will lead to confusing log messages for K8s nginx ingress users.

Reply:
@NickMeves K8s NGINX ingress with traefik is afaik not actually being the adviced way of running traefik anymore.
(as in: It is still formally "supported", but pretty much depricated)

Reply:
I meant for other non-traefik users -- If I use K8s nginx ingress with this PR using `rd`, this leads to a lot of superfluous and confusing log messages for me that aren't applicable.

I'd be tempted to say we just log what the final redirect URL is at the end once we settle on a mechanism to chose it.